test/v2/core/minter.spec.js

```diff
@@ -0,0 +1,212 @@
+    const splitterValues = [6, 3, 31];
+
+    const PaymentSplitter = await ethers.getContractFactory("PaymentSplitter");
+    paymentSplitter = await PaymentSplitter.deploy(splitterAddresses, splitterValues);
+    await paymentSplitter.deployed();
+
+    const rolloverVirtual = "1080000000000000000";
+    const vETH2Addr = "0x898bad2774eb97cf6b94605677f43b41871410b1";
+
+    const Withdrawals = await ethers.getContractFactory("Withdrawals");
+    withdrawals = await Withdrawals.deploy(vETH2Addr, rolloverVirtual);
+    await withdrawals.deployed();
+
+    const numValidators = 1000;
+    const adminFee = 0;
+
+    // const FeeCalc = await ethers.getContractFactory("FeeCalc");
+    // const feeCalc = await FeeCalc.deploy(parseEther("0"), parseEther("0"));
+    // await feeCalc.deployed();
+
+    const addresses = [
+      ethers.constants.AddressZero,
+      //feeCalc.address, // fee splitter
+      sgEth.address, // sgETH address
+      wsgEth.address, // wsgETH address
+      multiSig.address, // government address
+      ethers.constants.AddressZero, // deposit contract address - can't find deposit contract - using dummy address
+    ];
+
+    // add secondary minter contract / eoa
+    const Minter = await ethers.getContractFactory("SharedDepositMinterV2");
+    minter = await Minter.deploy(numValidators, adminFee, addresses);
+    await minter.deployed();
+
+    // const RewardsReceiver = await ethers.getContractFactory("RewardsReceiver");
+    // rewardsReceiver = await RewardsReceiver.deploy(withdrawals.address, [
+    //   sgEth.address,
+    //   wsgEth.address,
+    //   paymentSplitter.address,
+    //   minter.address,
+    // ]);
+    // await rewardsReceiver.deployed();
+
+    await sgEth.addMinter(minter.address);
+  });
+
+  it("deposit", async () => {
+    const prevBalance = await sgEth.balanceOf(alice.address);
+    await minter.connect(alice).deposit({
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(alice.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+  });
+
+  it("depositFor", async () => {
+    const prevBalance = await sgEth.balanceOf(alice.address);
+    await minter.depositFor(alice.address, {
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(alice.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+  });
+
+  it("depositAndStake", async () => {
+    const prevStake = await wsgEth.maxRedeem(deployer.address);
+    const prevBalance = await sgEth.balanceOf(wsgEth.address);
+    await minter.depositAndStake({
+      value: parseEther("1"),
+    });
+    const afterBalance = await sgEth.balanceOf(wsgEth.address);
+    expect(afterBalance).to.eq(prevBalance.add(parseEther("1")));
+
+    const afterStake = await wsgEth.maxRedeem(deployer.address);
+    expect(afterStake).to.eq(prevStake.add(parseEther("1")));
+  });
+
+  it("depositAndStakeFor", async () => {
```

Comment:
can we add a test to make sure this works for the direction case? 
i.e. Alice calls the function, but with Bobs address, and bob recvs the tokens in the end